pkg/pxc/app/statefulset/node.go

```diff
@@ -211,6 +225,62 @@ func (c *Node) SidecarContainers(spec *api.PodSpec, secrets string, cr *api.Perc
 	return nil, nil
 }
 
+func (c *Node) LogCollectorContainer(spec *api.LogCollectorSpec, secrets string, cr *api.PerconaXtraDBCluster) (*corev1.Container, error) {
+
+	logEnvs := []corev1.EnvVar{
+		{
+			Name:  "LOG_DATA_DIR",
+			Value: "/var/lib/mysql",
+		},
+		{
+			Name: "POD_NAMESPASE",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.namespace",
+				},
+			},
+		},
+		{
+			Name: "POD_NAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.name",
+				},
+			},
+		},
+	}
+
+	container := corev1.Container{
+		Name:            "log-collector",
+		Image:           spec.Image,
+		ImagePullPolicy: spec.ImagePullPolicy,
+		Env:             logEnvs,
+		SecurityContext: spec.ContainerSecurityContext,
+	}
+
+	res, err := app.CreateResources(spec.Resources)
+	if err != nil {
+		return nil, fmt.Errorf("create resources error: %v", err)
+	}
+	container.Resources = res
+
+	container.VolumeMounts = []corev1.VolumeMount{
+		{
+			Name:      DataVolumeName,
+			MountPath: "/var/lib/mysql",
+		},
+	}
+
+	if cr.Spec.LogCollector != nil && cr.Spec.LogCollector.Configuration != "" {
```

Comment:
can it be a secret?

Reply:
we will use secret for env variables which can be used as a secrets